ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/AggregateAnalyzer.java

```diff
@@ -344,8 +384,10 @@ public Void visitUnqualifiedColumnReference(
         final UnqualifiedColumnReferenceExp node,
         final Void context
     ) {
-      dereferenceCollector.accept(aggFunctionName, node);
-
+      if (!foundExpressionInGroupBy || visitedAggFunction) {
+        dereferenceCollector.accept(aggFunctionName, node);
+      }
+      foundExpressionInGroupBy = false;
```

Comment:
I'm a little confused here, why do we unset the `foundExpressioninGroupBy` here? I feel like we should set it right after we set it and call `super`. For example we should unset it after calling `super.visitSubscriptExpression(node, context);`